File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        // 1,2
        // 1,3,2
        // 1,3,2,3
        int res = 0;
        int l = 0;
        int r = heights.length - 1;
        int area;

        while (l < r) {
            area = Math.min(heights[l], heights[r]) * (r - l);
            res = Math.max(area, res);
            if (heights[r] < heights[l]) {
                r--;
            } else {
                l++;
            }
        }

        return res;


    }
}

File: Data Structures & Algorithms/pow-x-n/submission-0.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        res = 0
        if n == 0:
            return 1
        if n == 1:
            return x
        
        val = 1
        for i in range(abs(n)):
            val *= x
       
        if n > 1:
            return val
        elif n < 0:
            return 1 / val

        